presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveFileFormats.java

```diff
@@ -240,12 +240,17 @@ public void testRcTextOptimizedWriter(int rowCount)
                 .filter(TestHiveFileFormats::withoutNullMapKeyTests)
                 .collect(toImmutableList());
 
-        assertThatFileFormat(RCTEXT)
-                .withColumns(testColumns)
-                .withRowsCount(rowCount)
-                .withFileWriterFactory(new RcFileFileWriterFactory(HDFS_ENVIRONMENT, TYPE_MANAGER, new NodeVersion("test"), HIVE_STORAGE_TIME_ZONE, STATS))
-                .isReadableByRecordCursor(createGenericHiveRecordCursorProvider(HDFS_ENVIRONMENT))
-                .isReadableByPageSource(new RcFilePageSourceFactory(TYPE_MANAGER, HDFS_ENVIRONMENT, STATS, new HiveConfig()));
+        // the test writes longs, so we can't use nanos precision
+        for (HiveTimestampPrecision timestampPrecision : List.of(MILLISECONDS, MICROSECONDS)) {
```

Comment:
is NANOSECONDS covered elsewhere?

also, why testRcTextOptimizedWriter loops over precisions and eg testOrc does not?